pkg/plugins/golang/v2/api.go

```diff
@@ -175,9 +177,9 @@ func (p *createAPISubcommand) GetScaffolder() (cmdutil.Scaffolder, error) {
 		return nil, fmt.Errorf("unknown pattern %q", p.pattern)
 	}
 
-	// Create the actual resource from the resource options
-	res := p.resource.NewResource(p.config, p.doResource)
-	return scaffolds.NewAPIScaffolder(p.config, string(bp), res, p.doResource, p.doController, p.force, plugins), nil
+	// Create the resource from the options
+	res := p.options.NewResource(p.config)
+	return scaffolds.NewAPIScaffolder(p.config, string(bp), res, p.force, plugins), nil
```

Comment:
See my other comment. Options are not valid for all kinds. 
We need the : 

// Create the actual resource from the resource options
	res := p.resource.NewResource(p.config, p.doResource)